### ISSUE TITLE
${#arr[@]:-} array length substitution should be flagged as error - bash 4.4 will report "bad substitution" 

### ISSUE THREAD

**Issue by PeterPitterling** [koalaman/shellcheck#2386]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash    
declare -a arr=()
echo ${#arr[@]-}
echo ${#arr[@]:-}
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
warn
https://stackoverflow.com/questions/42766462/why-isnt-array-valid-in-bash-4-4-when-it-was-accepted-by-earlier-rele
